File: 905-sort-array-by-parity/905-sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
//         vector<int> v;
//         for(auto &i:nums){
//             if(i%2==0){
//                 v.push_back(i);
//             }
//         }
        
//         for(auto &i:nums){
//             if(i%2!=0){
//                 v.push_back(i);
//             }
//         }
        
//         return v;
        
        int i =0 , j=nums.size()-1;
        while(i<j){
            if(nums[i]%2==0){
                i++;
            }
            else if(nums[i]%2!=0){
                while(j>i && nums[j]%2!=0){
                    j--;
                }
                swap(nums[i],nums[j]);
            }
        }
        
        return nums;
    }
};

File: 905-sort-array-by-parity/905-sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int i =0;
        int j = 0;
        int n = nums.length;
        for(int k=0;k<nums.length;k++){
            if(nums[k]%2==0){
                swap(nums,i,k);
                i++;
            }
        }
        
        return nums;
    }
    
    public static void swap(int nums[],int i, int j){
        int temp = nums[i];
        nums[i]=nums[j];
        nums[j]=temp;
    }
}